bittensor/core/metagraph.py

```diff
@@ -597,12 +588,19 @@ def _initialize_subtensor(self, subtensor: "Subtensor"):
 
                 subtensor = self._initialize_subtensor(subtensor)
         """
+        if subtensor and subtensor != self.subtensor:
+            self.subtensor = subtensor
+        if not subtensor and self.subtensor:
+            subtensor = self.subtensor
         if not subtensor:
             # TODO: Check and test the initialization of the new subtensor
             # Lazy import due to circular import (subtensor -> metagraph, metagraph -> subtensor)
             from bittensor.core.subtensor import Subtensor
 
-            subtensor = Subtensor(network=self.network)
+            if self.network == "unknown" and self.chain_endpoint:
```

Comment:
```suggestion
            if self.chain_endpoint:
```
we don't want to require that self.network be passed the value of "unknown" here.  probably should just have chain_endpoint be used in preference over network.  or even just merge them and allow the self.network to be passed a chain endpoint directly, without the extra arg